both/methods.js

```diff
@@ -135,65 +128,52 @@ Meteor.methods({
     // in order, but I don't know if there is any guarantee of such across users
     // See http://docs.meteor.com/#method_unblock.
 
+    var newCount = Rounds.find({competitionId: competitionId, eventCode: eventCode}).count() + 1;
     var newRoundId = Rounds.insert({
       competitionId: competitionId,
       eventCode: eventCode,
       formatCode: wca.formatsByEventCode[eventCode][0],
-
-      // These will be filled in by refreshRoundCodes, but
-      // add valid value so the UI doesn't crap out.
-      roundCode: 'f',
-      nthRound: wca.MAX_ROUNDS_PER_EVENT,
+      nthRound: newCount,
+      totalRounds: newCount,
     });
 
+    Rounds.update(
+      {competitionId: competitionId, eventCode: eventCode},
```

Comment:
That's the price for making the field not optional.

I can space out the objects.